src/server.rs

```diff
@@ -553,14 +575,35 @@ impl Server {
         Ok(())
     }
 
+    pub async fn checkin_cleanup(&mut self) -> Result<(), Error> {
+        if self.in_transaction() {
+            self.query("ROLLBACK").await?;
+        }
+
+        if self.needs_cleanup {
+            self.query("DISCARD ALL").await?;
+            self.needs_cleanup = false;
+        }
+
+        self.set_name("pgcat").await?;
```

Comment:
I would skip that one tbh, 9 times out of 10 the same application will be back to use that connection. Saves one query on checkin, might be worth it?